settings.py

```diff
@@ -981,6 +983,12 @@
     from dev_settings import *
 
 
+if hasattr(COMPRESS_ENABLED, '__call__'):
+    COMPRESS_ENABLED = COMPRESS_ENABLED()
+if hasattr(COMPRESS_OFFLINE, '__call__'):
+    COMPRESS_OFFLINE = COMPRESS_OFFLINE()
```

Comment:
this is an interesting pattern. is it one that we use elsewhere or you got from somewhere?

it feels like it'd be clearer to just use booleans and rely on all settings overriding the defaults. but maybe that makes rollout more annoying?

Reply:
Yeah one of my goals was not to have to make any localsettings changes to existing environments, and for people's local dev environments to shift over to the new default dev setting, unless they have `COMPRESS_ENABLED` or `COMPRESS_OFFLINE` set in which case that's respected.

My lambda / __call__ shenanigans here raised my own "possibly too clever" flag, but it was either that or removing the default definitions from the block with the other COMPRESS_* defaults (where it feels like they should be grouped) and instead checking for their presence in `locals()` after `from localsettings import *` section, which to me felt more brittle and (even) harder to follow.

Reply:
That makes sense.

I think the other alternative would be to lift the goal of not requiring local changes in all scenarios (e.g. if you don't use `dev_settings`). I'd personally favor one-time-pain for long term clarity, but can see how it's subjective.